x-pack/plugin/ilm/qa/multi-node/src/javaRestTest/java/org/elasticsearch/xpack/ilm/actions/RolloverActionIT.java

```diff
@@ -193,14 +195,105 @@ public void testRolloverActionWithMaxPrimaryDocsSize() throws Exception {
         }, 30, TimeUnit.SECONDS);
     }
 
+    public void testRolloverActionWithEmptyIndex() throws Exception {
+        String originalIndex = index + "-000001";
+        String secondIndex = index + "-000002";
+        createIndexWithSettings(
+            client(),
+            originalIndex,
+            alias,
+            Settings.builder()
+                .put(IndexMetadata.SETTING_NUMBER_OF_SHARDS, 1)
+                .put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, 0)
+                .put(RolloverAction.LIFECYCLE_ROLLOVER_ALIAS, alias)
+        );
+
+        // create policy
+        createNewSingletonPolicy(
+            client(),
+            policy,
+            "hot",
+            new RolloverAction(null, null, TimeValue.timeValueSeconds(1), null, null, null, null, null, null, null)
+        );
+        // update policy on index
+        updatePolicy(client(), originalIndex, policy);
+
+        // maybe set the controlling setting to explicitly true (rather than just true by default)
+        if (randomBoolean()) {
+            setLifecycleRolloverOnlyIfHasDocumentsSetting(true);
+        }
+
+        // because the index is empty, it doesn't roll over
+        assertBusy(() -> {
+            assertThat(getStepKeyForIndex(client(), originalIndex).getName(), is(WaitForRolloverReadyStep.NAME));
+            assertFalse(indexExists(secondIndex));
+            assertTrue(indexExists(originalIndex));
+        }, 30, TimeUnit.SECONDS);
+
+        switch (between(0, 2)) {
```

Comment:
I think rather than randomize between the three potential ways to force a rollover, I'd rather explicitly do each of them. This would allow us to catch issues better in the future that might only break one of the branches since we don't test each method every CI run.

Reply:
How does 1ea80c1bbcadf016801a9d09a5cb9b50f49b1e12 work for you, @dakrone?

Reply:
Works for me, 🚢